src/main/java/org/commcare/data/xml/VirtualInstances.java

```diff
@@ -19,24 +19,27 @@ public class VirtualInstances {
     public static final String SELCTED_CASES_INSTANCE_ROOT_NAME = "results";
     public static final String SELCTED_CASES_INSTANCE_NODE_NAME = "value";
 
-    public static VirtualDataInstance buildSearchInputInstance(Map<String, String> userInputValues) {
+    public final static String JR_SEARCH_INPUT_REFERENCE = "jr://instance/search_input";
+    public final static String JR_SELECTED_VALUES_REFERENCE = "jr://instance/selected_cases";
```

Comment:
yes that make sense to me, how about sticking to `jr://instance/remote` in that case since we already have that one utilised heavily in current apps ?

Reply:
And confirming if it matches your understanding that instanceId check is suffcient and `stepType` check [here](https://github.com/dimagi/commcare-core/blob/refactorMultiSelect/src/main/java/org/commcare/core/process/CommCareInstanceInitializer.java#L204) is redundant

Reply:
> yes that make sense to me, how about sticking to jr://instance/remote in that case since we already have that one utilised heavily in current apps ?

I think it makes sense to use a separate reference since it does communicate something about the instance. `remote` reference are fetched from and external system while `virtual` ones are created during session navigation. I don't think there's a downside to introducing a new one.

I have also been thinking about if we can standardise the 'virtual' instance schema or otherwise communicate what the expected schema is. With 'remote' instances the schema is always the same (at least so far). But with these 'virtual' instances we already have 2 different schemas. This is probably a problem for another time but I was thinking of adding an optional 'schema' attribute to the instance def which could be used to indicate what the expected schema should be.


> And confirming if it matches your understanding that instanceId check is suffcient and `stepType` check [here](https://github.com/dimagi/commcare-core/blob/refactorMultiSelect/src/main/java/org/commcare/core/process/CommCareInstanceInitializer.java#L204) is redundant

Yes, if necessary we could prevent adding instances into the stack with duplicate IDs.

Reply:
that distinction makes sense to me and I am happy with using `virtual`.

Though your second point makes me think there is value in keeping these as separate references because they have different schemas. Like why not use the reference itself as an indication of a distinct schema it represents instead of adding a separate field to the instance ?

Reply:
> Though your second point makes me think there is value in keeping these as separate references because they have different schemas. Like why not use the reference itself as an indication of a distinct schema it represents instead of adding a separate field to the instance ?

Maybe, though I still think it would be nice to namespace them.